ui/src/shared/components/time_series/TimeSeries.tsx

```diff
@@ -153,7 +163,9 @@ class TimeSeries extends Component<Props, State> {
       return
     }
 
-    this.executeQueries()
+    if (this.isComponentMounted) {
+      this.executeQueries()
```

Comment:
any race conditions if this is not `await`ed?

Reply:
the race conditions are being addressed by the latestUUID property on the class